extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/static_resource/StaticResourceBuildTimeConfig.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.vertx.http.runtime.static_resource;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "static-resource", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class StaticResourceBuildTimeConfig {
+    /**
```

Comment:
I think I would make this a Map of potential endpoints to support several. But then I don't know if the Map is properly ordered. We need to make sure things are ordered so I would use a `SortedMap` and check the config code handles that properly.

Reply:
I am thinking of having something like this for build time configuration. 

```
    /**
     * The endpoint from which the static resources will be served.
     */
    @ConfigDocMapKey("endpoint")
    @ConfigItem(name = ConfigItem.PARENT)
    public SortedMap<String, StaticResourceGeneralConfig> config;

    @ConfigGroup
    public static class StaticResourceGeneralConfig {
         @ConfigItem(name=ConfigItem.PARENT)
        public boolean enabled; 

        @ConfigItem
        public boolean enableCache; 
        
        @ConfigItem
        public Optional<MemorySize> cacheSize;
        
        @ConfigItem
        public Optional<String> index;
        
        @ConfigItem
        public boolean allowDirectoryListing;
    }
```

WDYT? 

The directory path for each endpoint can be made a runtime config to be overridable through env variable - thanks to @rsvoboda for the initial idea.

Reply:
> I added a few more comments.

Thanks @gsmet let me check them.